### PR TITLE
[W5.10h][T11-A4]Eka Buyung Lienadi

### PR DESCRIPTION
**Pull request by yungyung04**
create Use Cases

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -88,6 +88,44 @@ Use case ends.
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
 
+
+#### Use case: Rename the tag 'friends' into 'buddies'
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:92]: This use case is too specific, you can change it to something like `Rename tags`.

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -88,6 +88,44 @@ Use case ends.
...
+> 4.   AddressBook asks a new name for the tag
+> 4a1. User cancels the request.
+  Use case resumes from step 3
+
+3b. User provides an invalid input
+
+> 3b1. AddressBook shows an error message
+  Use case ends
+
+5a. User provides an unacceptable tagname
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:117]: what is an `unacceptable tagname`? need to be more specific here.

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -88,6 +88,44 @@ Use case ends.
...
+
+> 5a1. AddressBook shows an error message
+  Use case ends
+
+5b. User makes a typo
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:122]: You can be more specific on defining what a `typo` is.

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -88,6 +88,44 @@ Use case ends.
...
+Precondition : 'friends' tag exists in the address book
+**MSS**
+
+1. User requests to list all tags
+2. AddressBook shows a list of all tags
+3. User requests to rename the 'friends' tag
+4. AddressBook asks a new name for the tag
+5. User responds by typing 'buddies'
+6. AddressBook displays a message indicating the tag has been renamed
+Use case ends
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:103]: Can remove example tags, e.g. `friends` and `buddies`.

File: doc/DeveloperGuide.md [hunk 5]
```diff
@@ -88,6 +88,44 @@ Use case ends.
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
 
+
+#### Use case: Rename the tag 'friends' into 'buddies'
+
+Precondition : 'friends' tag exists in the address book
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:94]: Can be generalized to `some tags exists ...`

File: doc/DeveloperGuide.md [hunk 6]
```diff
@@ -88,6 +88,44 @@ Use case ends.
...
+3. User requests to rename the 'friends' tag
+4. AddressBook asks a new name for the tag
+5. User responds by typing 'buddies'
+6. AddressBook displays a message indicating the tag has been renamed
+Use case ends
+
+**Extensions**
+3a. User chooses a wrong tag to rename
+
+> 4.   AddressBook asks a new name for the tag
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:108]: Should be 3a1.

File: doc/DeveloperGuide.md [hunk 7]
```diff
@@ -88,6 +88,44 @@ Use case ends.
...
+  Use case ends
+
+5a. User provides an unacceptable tagname
+
+> 5a1. AddressBook shows an error message
+  Use case ends
+
+5b. User makes a typo
+
+> 6. AddressBook displays a message indicating the tag has been renamed
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md:124]: Should be 6b1

**Comment by jeffryhartanto** [comment]
@yungyung04 
Some comments added. Please close the PR after reading comments.

File: doc/DeveloperGuide.md (hunk 6 above)
Comment (yungyung04) [reply doc/DeveloperGuide.md:108]: should I still write 3a1. even though the step is exactly the same as step no.4 in MSS?

Comment (jeffryhartanto) [reply doc/DeveloperGuide.md:108]: Yes.. You can think of this extension as different scenario, so the numbering should be consistent.

Comment (yungyung04) [reply doc/DeveloperGuide.md:108]: Got it. Thx!

**Pull request closed without merging**